### PR TITLE
Remove migration code for Plutono and Vali

### PR DESCRIPTION
**Pull request by rickardsjp**
**How to categorize this PR?**

/area monitoring
/kind cleanup

**What this PR does / why we need it**:
#7318 introduced migration code to switch from Grafana to Plutono, and from Loki to Vali.
The commits that introduced these changes can't simply be reverted, so this commit removes the migration code.

Please note that artifacts from control planes which have been broken continuously since gardener [v1.71.0](https://github.com/gardener/gardener/releases/tag/v1.71.0) where the migration was introduced may need to be cleaned up manually. These artifacts can include ConfigMaps, Ingresses, PersistentVolumes, PersistentVolumeClaims, Deployments, and StatefulSets. The below commands can be used to clean up these artifacts in a control plane namespace in a seed:
```
kubectl delete statefulsets loki
kubectl delete persistentvolumeclaims loki-loki-0
kubectl delete deployments grafana
kubectl delete services grafana loki
kubectl delete configmaps -l app=loki
kubectl delete configmaps -l component=grafana
kubectl delete secret shoot-access-promtail
kubectl delete secret -l name=loki-tls
kubectl delete ingresses grafana loki
kubectl delete hvpa loki
kubectl get networkpolicies | grep -E 'loki|grafana' | awk '{print $1}' | xargs kubectl delete networkpolicies
```

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
/cc @istvanballok 

**Release note**:

```breaking operator
Migration code for Plutono and Vali is now removed. Consider manual cleanup for longterm broken Shoots as described in the PR to avoid leaking Loki's PV.
```

### CODE REVIEW

**Review by shafeeqes — COMMENTED** [review]
Thanks for cleaning up.
Do we need to adapt https://github.com/gardener/gardener/blob/67a049a3f66ce489002a67dd59ac7b95e8d2573b/pkg/component/plutono/plutono.go#L880 as well?

**Comment by shafeeqes** [comment]
> Thanks for cleaning up. Do we need to adapt
> 
> https://github.com/gardener/gardener/blob/67a049a3f66ce489002a67dd59ac7b95e8d2573b/pkg/component/plutono/plutono.go#L880
> 
> as well?

Okay, this is for the extensions. I see that they still use Grafana and loki instead of Plutono and vali.
Then it's fine.

**Review by istvanballok — CHANGES_REQUESTED** [review]
The base commit of the PR is a bit old, so please consider rebasing it to the master branch, just to make sure that we remove the Grafana/Loki terms that might have been added in the meantime.

```
$ git log HEAD^ -1 | grep Date
Date:   Thu Dec 7 16:18:43 2023 +0530
$ git log --oneline HEAD..origin/master | wc -l
39
```

I checked that there are no new such strings, but a rebase could still be worthwhile.

I checked for the occurrences of the Grafana, Loki and promtail strings, excluding some paths that should not be updated (e.g. docs/proposals)

```
$ git grep -l -i -E 'loki|grafana|promtail|istvan|jeremy' :^docs/proposals :^docs/usage/images :^pkg/component/plutono/plutono.go :^pkg/component/logging/fluentoperator/ ':^example/seed-crds/10-crd-fluentbit.fluent*' :^test/framework/applications/charts/redis/README.md :^pkg/component/kubestatemetrics
example/gardener-local/gardenlet/values-ipv6.yaml
pkg/component/plutono/testdata/configmap.yaml
test/testmachinery/shoots/logging/utils.go
```

Maybe we could adjust those files as well, and replace the recently added terms of Grafana and Loki with Plutono and Vali.

File: pkg/component/logging/vali/vali.go [hunk 1]
```diff
@@ -133,7 +133,6 @@ type Values struct {
 	CuratorImage          string
 	TelegrafImage         string
 	KubeRBACProxyImage    string
```
Comment (istvanballok) [inline pkg/component/logging/vali/vali.go]: This block needs to be formatted after this change.

File: pkg/component/logging/vali/vali_test.go [hunk 2]
```diff
@@ -145,7 +145,6 @@ var _ = Describe("Vali", func() {
 					},
 					ValiImage:             valiImage,
 					CuratorImage:          curatorImage,
-					RenameLokiToValiImage: alpineImage,
```
Comment (istvanballok) [inline pkg/component/logging/vali/vali_test.go:148]: and also the blocks in this file

**Review by istvanballok — CHANGES_REQUESTED** [review]

**Review by istvanballok — CHANGES_REQUESTED** [review]
`pkg/operation/common` needs to be removed from `skaffold.yaml` now that the package is removed - otherwise the `make gardener-up` step fails.

File: pkg/component/logging/vali/vali.go (hunk 1 above)
Comment (rickardsjp) [reply pkg/component/logging/vali/vali.go]: Yes, `make generate` takes care of that. Thank you.

**Review by istvanballok — COMMENTED** [review]
The change looks good to me. I double checked the tests and shoot/seed creation in the local setup.

I think we would need to add some hints (and optionally some instructions) to the PR description on how long time broken shoot control planes that could not be reconciled and hence were not migrated could be cleaned up manually. Manually deleting the loki and grafana artifacts is needed if such long time broken shoots exist in a Gardener landscape because otherwise, now that the migration coding is removed, if these shoots are fixed, the grafana and loki artifacts (especially pvc) would stay around as a leftover.

**Review by istvanballok — APPROVED** [review]
/lgtm

File: pkg/component/logging/vali/vali_test.go [hunk 3]
```diff
@@ -465,43 +462,6 @@ var _ = Describe("Vali", func() {
 			ctrl.Finish()
 		})
 
-		It("should patch garden/loki's PVC when new size is greater than the current one", func() {
```
Comment (ialidzhikov) [inline pkg/component/logging/vali/vali_test.go:468]: Why is this unit test being removed? It seems to be still a valid unit test and I was wondering why we need to remove it as part of this PR.

Comment (rickardsjp) [reply pkg/component/logging/vali/vali_test.go:468]: Oof, great catch. Thank you.

**Review by ialidzhikov — COMMENTED** [review]
Code-wise
/lgtm

I will approve after we agree internally how we clean up the leftover loki PVCs for broken Shoots since longtime.

**Review by rfranzke — COMMENTED** [review]
Hi folks,
what's the status of this PR? Can it be approved and merged?

**Comment by ialidzhikov** [comment]
> I will approve after we agree internally how we clean up the leftover loki PVCs for broken Shoots since longtime.

With @rfranzke we agreed to follow-up internally on the cleanup and meanwhile to proceed with this PR.

/approve

**Pull request merged**